internal/tasks/image_gc.go

```diff
@@ -227,40 +232,43 @@ func (j *Janitor) evaluatePolicy(ctx context.Context, proc *processor.Processor,
 
 		// track matching "delete" policies in GCStatus to allow users insight
 		// into how policies match
-		if policy.Action == "delete" {
-			m.GCStatus.RelevantPolicies = append(m.GCStatus.RelevantPolicies, policy)
+		if gcPolicies.Action == "delete" {
+			m.GCStatus.RelevantPolicies = append(m.GCStatus.RelevantPolicies, gcPolicies)
 		}
 
 		// evaluate constraints
-		if !policy.MatchesTags(m.TagNames) {
+		if !gcPolicies.MatchesTags(m.TagNames) {
 			continue
 		}
-		if !policy.MatchesTimeConstraint(m.Manifest, aliveManifests, j.timeNow()) {
+		if !gcPolicies.MatchesTimeConstraint(m.Manifest, aliveManifests, j.timeNow()) {
 			continue
 		}
 
-		pCopied := policy
+		pCopied := gcPolicies
 		// execute policy action
-		switch policy.Action {
+		switch gcPolicies.Action {
 		case "protect":
 			m.GCStatus.ProtectedByPolicy = &pCopied
 		case "delete":
-			err := proc.DeleteManifest(ctx, account, repo, m.Manifest.Digest, keppel.AuditContext{
+			err := proc.DeleteManifest(ctx, account, repo, m.Manifest.Digest, tagPolicies, keppel.AuditContext{
 				UserIdentity: janitorUserIdentity{
 					TaskName: "policy-driven-gc",
 					GCPolicy: &pCopied,
 				},
 				Request: janitorDummyRequest,
 			})
+			if _, ok := errext.As[processor.DeleteManifestBlockedByTagPolicyError](err); ok { //nolint:errcheck // intended to be skipped
+				continue
+			}
```

Comment:
The error should not be swallowed. As described in my previous review notes, type keppel.GCStatus should be extended to report the relevant tag policy.